PLUGIN_STYLEGUIDE.md

```diff
@@ -0,0 +1,146 @@
+# Contents
+---
+- [Objective](#objective)
+- [Format](#format)
+- [Structure](#structure)
+    - [Overview](#overview)
+    - [Files](#files)
+    - [Usage Examples](#usage-examples)
+    - [Configuration](#configuration)
+        - [Asset registration](#asset-registration)
+        - [Asset definition](#asset-definition)
+        - [Resource definitions](#resource-definitions)
+    - [Installation from source](#installation-from-source)
+    - [Additional notes](#additional-nodes)
+    - [Contributing](#contributing)
+- [Plugin collections](#plugin-collections)
+- [Tables of contents](#tables-of-contents)
+
+# Objective
+
+Thank you for your interest in contributing to the Sensu community! Contributing plugins is a great way to get involved, and we want to help you effectively explain  how to use your plugin.  
+
+This guide is designed to help you and other Sensu plugin authors describe your plugin’s use and facilitate consistency among community plugin READMEs. Because many users throughout the Sensu community rely on plugins, we value plugin documentation that:
+
+* Is readable and free of jargon
+* Focuses on users
+* Clearly describes what the plugin does
+* Includes examples of how to use and deploy the plugin
+
+# Format
+
+Plugin READMEs are Markdown documents. Write your README in [GitHub-flavored Markdown](https://guides.github.com/features/mastering-markdown/) to ensure it is properly rendered on Bonsai. 
+
+# Structure
+
+Organize your plugin README according to this overall structure:
+
+* Overview
+* Files
+* Usage examples
+* Configuration
+  * Asset registration
+  * Asset configuration
+  * Resource definitions
+* Installation from source
+* Contributing
+* Additional notes
+
+
+## Overview
+
+Begin with an overview of the plugin’s functionality. Describe what the plugin does and what it’s used for. Here are some examples of good overviews from Sensu plugins:
+
+
+
+*   [Prometheus Collector](https://github.com/sensu/sensu-prometheus-collector#overview)
+*   [Linux System Profiler](https://github.com/sensu/system-profile-linux#overview)
+*   [Influxdb Handler](https://github.com/sensu/sensu-influxdb-handler#sensu-influxdb-handler)
+*   [Fatigue Check Filter](https://github.com/nixwiz/sensu-go-fatigue-check-filter#sensu-go-fatigue-check-filter)
+*   [Has Contact Filter](https://github.com/sensu/sensu-go-has-contact-filter#sensu-go-has-contact-filter)
+
+
+## Files
+
+If the project is a collection (grouping of multiple plugins), the README should include a list of executable plugin files. If the project is not a collection, you do not need to include a list of files.
+
+
+## Usage examples
+
+A plugin’s usage is of primary concern when writing plugin documentation. The usage section should show the help output for a plugin, if available. Otherwise, any supported configuration via command-line arguments or environment variables should be thoroughly documented.
```

Comment:
```suggestion
Plugin documentation should include a usage section, which should show the help output for the plugin, if available. Otherwise, any supported configuration via command-line arguments or environment variables should be thoroughly documented in the usage section.
```